tests/cmdline/commands/test_storage.py

```diff
@@ -103,3 +103,52 @@ def mocked_migrate(self):  # pylint: disable=no-self-use
     assert result.exc_info[0] is SystemExit
     assert 'Critical:' in result.output
     assert 'passed error message' in result.output
+
+
+def tests_storage_maintain_full_dry(run_cli_command):
+    """Test that ``verdi storage migrate`` detects the cancelling of the interactive prompt."""
+    result = run_cli_command(cmd_storage.storage_maintain, options=['--full', '--dry-run'], raises=True)
+    assert '--dry-run' in result.output.lower()
+    assert '--full' in result.output.lower()
+
+
+def tests_storage_maintain_logging(run_cli_command, monkeypatch, caplog):
+    """Test all the information and cases of the storage maintain command."""
+    import logging
+
+    from aiida.backends import control
+
+    def mock_maintain(**kwargs):
+        logmsg = 'Provided kwargs:\n'
+        for key, val in kwargs.items():
+            logmsg += f' > {key}: {val}\n'
+        logging.info(logmsg)
+
+    monkeypatch.setattr(control, 'repository_maintain', mock_maintain)
+
+    with caplog.at_level(logging.INFO):
+        _ = run_cli_command(cmd_storage.storage_maintain, user_input='Y')
```

Comment:
Might as well not assign the result if you are not going to use it
```suggestion
        run_cli_command(cmd_storage.storage_maintain, user_input='Y')
```